code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm

```diff
@@ -1,43 +1,101 @@
+	to_chat(owner, "<span class='xenodanger'>You feel your accelerant glands refill. You can use Defile again.</span>")
 	return ..()
 
-/datum/action/xeno_action/activable/larval_growth_sting/defiler/use_ability(atom/A)
+/datum/action/xeno_action/activable/defile/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return
+
+	if(!iscarbon(A))
+		if(!silent)
+			to_chat(owner, "<span class='xenodanger'>Our sting won't affect this target!</span>")
+		return FALSE
+
+	var/mob/living/carbon/victim = A
+
+	if(!victim?.can_sting())
+		if(!silent)
+			to_chat(owner, "<span class='xenodanger'>Our sting won't affect this target!</span>")
+		return FALSE
+
+	if(!owner.Adjacent(victim))
+		var/mob/living/carbon/xenomorph/X = owner
+		if(!silent)
+			to_chat(X, "<span class='warning'>We can't reach this target!</span>")
+		return FALSE
+
+/datum/action/xeno_action/activable/defile/use_ability(atom/A)
 	var/mob/living/carbon/xenomorph/Defiler/X = owner
-	var/mob/living/carbon/C = A
-	if(locate(/obj/item/alien_embryo) in C) // already got one, stops doubling up
-		return ..()
-	if(!do_after(X, DEFILER_STING_CHANNEL_TIME, TRUE, C, BUSY_ICON_HOSTILE))
+	var/mob/living/carbon/living_target = A
+	X.face_atom(living_target)
+	if(!do_after(X, DEFILER_DEFILE_CHANNEL_TIME, TRUE, living_target, BUSY_ICON_HOSTILE))
+		add_cooldown(DEFILER_FAIL_COOLDOWN)
 		return fail_activate()
 	if(!can_use_ability(A))
 		return fail_activate()
 	add_cooldown()
-	X.face_atom(C)
-	X.do_attack_animation(C)
-	playsound(C, pick('sound/voice/alien_drool1.ogg', 'sound/voice/alien_drool2.ogg'), 15, 1)
-	var/obj/item/alien_embryo/embryo = new(C)
-	embryo.hivenumber = X.hivenumber
-	GLOB.round_statistics.now_pregnant++
-	SSblackbox.record_feedback("tally", "round_statistics", 1, "now_pregnant")
-	to_chat(X, "<span class='xenodanger'>Our stinger successfully implants a larva into the host.</span>")
-	to_chat(C, "<span class='danger'>You feel horrible pain as something large is forcefully implanted in your thorax.</span>")
-	C.apply_damage(100, STAMINA)
-	C.apply_damage(10, BRUTE, "chest", updating_health = TRUE)
-	C.emote("scream")
+	X.face_atom(living_target)
+	X.do_attack_animation(living_target)
+	playsound(living_target, 'sound/effects/spray3.ogg', 15, TRUE)
+	playsound(living_target, pick('sound/voice/alien_drool1.ogg', 'sound/voice/alien_drool2.ogg'), 15, 1)
+	to_chat(X, "<span class='xenodanger'>Our stinger successfully discharges accelerant into our victim.</span>")
+	to_chat(living_target, "<span class='danger'>You feel horrible pain as something sharp forcibly pierces your thorax.</span>")
+	living_target.apply_damage(50, STAMINA)
+	living_target.apply_damage(5, BRUTE, "chest", updating_health = TRUE)
+	living_target.emote("scream")
+
+	var/defile_strength_multiplier = 1
+	var/defile_reagent_amount
+	var/defile_power
+
+	if(!LAZYLEN(GLOB.defile_purge_list))
+		return
```

Comment:
don't really get the point of this

Reply:
Sanity check; I can remove it though, I don't care